packages/effect/src/internal/stream.ts

```diff
@@ -8045,3 +8045,28 @@ export const encodeText = <E, R>(self: Stream.Stream<string, E, R>): Stream.Stre
     const encoder = new TextEncoder()
     return map(self, (s) => encoder.encode(s))
   })
+
+/** @internal */
+export const fromEventListener = (
```

Comment:
I think we can add a generic here for overriding the event type, with a default of `Event`.

Reply:
You can then use that to replace the as any casts in platform.

Reply:
Ah yeah, that's better. :)